src/Symfony/Bundle/FrameworkBundle/Resources/config/json_streamer.php

```diff
@@ -48,53 +49,62 @@
             ->args([
                 service('type_info.resolver'),
             ])
-        ->set('.json_streamer.write.property_metadata_loader.generic', GenericTypePropertyMetadataLoader::class)
+        ->set('.json_streamer.write.property_metadata_loader.attribute', WriteAttributePropertyMetadataLoader::class)
             ->decorate('json_streamer.write.property_metadata_loader')
             ->args([
                 service('.inner'),
-                service('type_info.type_context_factory'),
+                tagged_locator('json_streamer.value_transformer'),
+                service('type_info.resolver'),
             ])
-        ->set('.json_streamer.write.property_metadata_loader.date_time', WriteDateTimeTypePropertyMetadataLoader::class)
+        ->set('.json_streamer.write.property_metadata_loader.generic', GenericTypePropertyMetadataLoader::class)
             ->decorate('json_streamer.write.property_metadata_loader')
             ->args([
                 service('.inner'),
+                service('type_info.type_context_factory'),
             ])
-        ->set('.json_streamer.write.property_metadata_loader.attribute', WriteAttributePropertyMetadataLoader::class)
```

Comment:
If the order of decorators is meaningful, this should better be done using decoration priorities than changing the order of declarations in this file (which are not _guaranteed_ to be preserved if some compiler passes redefine services). And it will make it clear that there is an intent about the decoration order.

Reply:
another benefit of priorities (if we follow our usual convention of leaving gaps between the priorities used in core) is that it allows bundle to add an additional decorate between them if needed.